### PR TITLE
#1690 fix bus state change for node-breaker

### LINKED ISSUE #1690: Issue with calculated bus disconnection
Currently, we try to simulate the event by changing the "state" attribute through the Network model:

<dyn:blackBoxModel id="..." lib="EventSetPointReal" parFile="..." parId="..."/>
<dyn:connect id1="..." var1="event_state1" id2="NETWORK" var2="<bbs-id>_state"/>

with the parameters for the event:

<par name="event_stateEvent1" type="DOUBLE" value="1"/>
<par name="event_tEvent" type="DOUBLE" value="80"/>

To check the results we added a "timeline" and "curves" entry in the jobs file. If we look at the timeline output, it seems that the change in state has been taken into account:

80 | calculatedBus__0460f448-c766-11e1-8775-005056c00008_0 | BUS : switch off

But if we look at the curves file, where we have output the voltage of the related calculated bus, we see something weird:

time;NETWORK_calculatedBus__0460f448-c766-11e1-8775-005056c00008_0_Upu_value;
0.000000;0.955203;
10.000000;0.953801;
20.000000;0.952911;
30.000000;0.952353;
40.000000;0.952007;
50.000000;0.951794;
60.000000;0.951662;
70.000000;0.951581;
80.000000;0.000000;
80.000000;0.951533;
90.000000;0.951503;
100.000000;0.951484;
 

The voltage goes to zero at t=80 (the time of the event), but in the same t, it is "recovered". In the dynawo log we see the following (the bus is switched ON after being switched OFF):

INFO |       60.000 |        6               12          1             10.000
INFO |       70.000 |        7               14          1             10.000
DEBUG | keep SubNetworks num : 0
INFO | switch ON bus : calculatedBus__0460f448-c766-11e1-8775-005056c00008_0
DEBUG | Algebraic mode (with J recalculation) change for model NETWORK at t = 80
INFO | Algebraic mode (with J recalculation) change at t = 80
INFO |       80.000 |        8               15          1             10.000
INFO | calculation of the new starting point of the simulation.
DEBUG | call of SolverReInit ie a new symbolic and numerical factorization will be performed
INFO |       80.000 |        8               16          2             10.000
DEBUG | largest derivative values (truncated to the first 10) at the current iteration
...
DEBUG | call of SolverReInit ie a new symbolic and numerical factorization will be performed
INFO |       90.000 |        9               17          3             10.000
INFO |      100.000 |       10               18          3             10.000

### PR DESCRIPTION
**Pull request by marcochiaramello**
Closes #1690

### CODE REVIEW

File: dynawo/sources/Common/Dictionaries/DYNError_en_GB.dic [hunk 1]
```diff
@@ -184,6 +184,7 @@ VoltageLevelGraphUndefined  =             VOLTAGE LEVEL %1% : graph is undefined
 VoltageLevelBBSError        =             VOLTAGE LEVEL %1% : error while looking for the closest bus bar section
 FrequencyCollapse           =             frequency collapse (%1% Hz < %2% Hz)
 FrequencyIncrease           =             high frequency (%1% Hz > %2% Hz)
+CalculatedBusNoSwitchStateChange =        calculated bus %1% cannot change state as there are no retained switches close to
```
Comment (rosiereflo) [inline dynawo/sources/Common/Dictionaries/DYNError_en_GB.dic:187]: I would propose "calculated bus %1% cannot change state as there is no retained switch in the node"

**Review by rosiereflo — CHANGES_REQUESTED** [review]

File: dynawo/sources/Common/Dictionaries/DYNError_en_GB.dic (hunk 1 above)
Comment (marcochiaramello) [reply dynawo/sources/Common/Dictionaries/DYNError_en_GB.dic]: ok, done

**Review by rosiereflo — APPROVED** [review]

**Comment by marcochiaramello** [comment]
@rosiereflo do you think it can be useful to add an nrt for this specific case?

**Comment by rosiereflo** [comment]
> @rosiereflo do you think it can be useful to add an nrt for this specific case?

As it is already well covered by unit tests I'm not sure it is useful

File: dynawo/sources/Models/CPP/ModelNetwork/DYNModelBus.h [hunk 2]
```diff
@@ -625,6 +625,7 @@ class ModelBus : public NetworkComponent::Impl {  ///< Generic AC network bus
   std::vector<boost::weak_ptr<ModelSwitch> > connectableSwitches_;  ///< switch connected or connectable on the node
 
   const std::string modelType_;  ///< model Type
+  const bool isNodeBreaker_;  ///< true if bus is a calculated one
```
Comment (adriguir) [inline dynawo/sources/Models/CPP/ModelNetwork/DYNModelBus.h:628]: I would suggest the following comment: true if the bus is in node breaker (= calculated bus) or vice versa.

**Review by adriguir — CHANGES_REQUESTED** [review]

File: dynawo/sources/Models/CPP/ModelNetwork/DYNModelBus.h (hunk 2 above)
Comment (marcochiaramello) [reply dynawo/sources/Models/CPP/ModelNetwork/DYNModelBus.h]: ok, thanks

**Review by adriguir — APPROVED** [review]

**Pull request merged**